File: pat1061.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        String[] week = {"MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"};
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        String line1 = reader.readLine();
        String line2 = reader.readLine();
        String line3 = reader.readLine();
        String line4 = reader.readLine();
        int len1 = Math.min(line1.length(), line2.length());
        int len2 = Math.min(line3.length(), line4.length());
        int breakIndex = 0;     // 第一次遇到大写字母相等时的索引
        for (int i = 0; i < len1; i++) {
            char c1 = line1.charAt(i);
            char c2 = line2.charAt(i);
            if (c1 == c2 && c1 >= 'A' && c1 <= 'G') {
                System.out.print(week[c1 - 'A']);
                breakIndex = i + 1;
                break;
            }
        }

        for (int i = breakIndex; i < len1; i++) {
            char c1 = line1.charAt(i);
            char c2 = line2.charAt(i);
            if (c1 == c2) {
                if (Character.isDigit(c1)) {
                    System.out.printf(" %02d:", c1 - '0');
                    break;
                }
                if (c1 >= 'A' && c1 <= 'N') {
                    System.out.printf(" %02d:", c1 - 'A' + 10);
                    break;
                }

            }
        }

        for (int i = 0; i < len2; i++) {
            char c1 = line3.charAt(i);
            char c2 = line4.charAt(i);
            if (c1 == c2 && Character.isLetter(c1)) {
                System.out.printf("%02d", i);
                break;
            }

        }
    }
}

File: pat1061.cpp
#include<iostream>
#include<string>
using namespace std;

int main() {

	string week[] = { "MON","TUE","WED","THU","FRI","SAT","SUN" };
	string s1, s2, s3, s4;
	cin >> s1 >> s2 >> s3 >> s4;
	int len1 = s1.length();
	int len2 = s3.length();
	bool isWeek = true;	// 标志找到了星期几
	for (int i = 0; i < len1; i++) {
		if (s1[i] == s2[i] && s1[i] >= 'A'&&s2[i] <= 'G'&&isWeek == true) {
			cout << week[s1[i] - 'A'] << ' ';
			isWeek = false;
			continue;
		}
		if (s1[i] == s2[i] && isWeek == false) {
			if (s1[i] >= 'A'&&s1[i] <= 'N') {
				cout << (10 + s1[i] - 'A');
				break;
			}
			if (s1[i] >= '0'&&s1[i] <= '9') {
				cout << '0' << s1[i];
				break;
			}
		}
	}

	for (int i = 0; i < len2; i++) {
		if (s3[i] == s4[i] ) {
			if (s3[i] >= 'A'&&s3[i] <= 'Z' || s3[i] >= 'a'&&s3[i] <= 'z') {
				if (i < 10) {
					cout << ":0" << i;
				}
				else {
					cout << ':' << i;
				}
				break;
			}
		}
	}

	return 0;
}